### DISCUSSION TITLE
Always fetch

### DISCUSSION THREAD

**Discussion by nanandn** [TanStack/query#2784] [Q&A]
Is there a way to always fetch. I tried setting staleTime to 0 and cache time to 0. 

My use case is... I have a drop-down menu that is populated with data from the server. I want to fetch this every time the drop-down menu is open. I use queryClient.fetchQuery. 

Thanks
Anand

**Comment by TkDodo** [comment]
Disable the query when the menu is closed and enable it again when the menu opens. Alternatively, make the menu mount / unmount because that should also trigger a refetch.

**Reply by nanandn** [reply]
@TkDodo Thanks. It works now. There was a bug in my code. The menu does unmount when closing and I have cacheTime set to 0. Works now after I fixed the bug in my code. I can't exactly remember what I had to fix :-)

On a slightly related note...  Is there a way to tell react-query to always fetch but return the data from the cache first - like "cache-and-network" strategy in apollo. keepPrevious data documentation says "If set, any previous data will be kept when fetching new data because the query key changed.". But in this case the query is not changing.

Thanks
Anand

**Reply by TkDodo** [reply]
Not sure how that policy works, but there is nothing that lets you fetch on every re-render. I don’t think anybody would want that. The default staleTime of zero makes sure that every query is instantly stale and will be refetched at the next “smart” moment.
